File: Arrays_and_Hashing/int_sum.js
/**
 * @function
 * @param {number[]} nums 
 * @param {number} target 
 * @return {number[]}
 */
function twoSum(nums, target) {
  const num_index = new Map();

  for (let i = 0; i < nums.length; i++) {
    let difference = target - nums[i]
    if (num_index.has(difference)) {
      return [num_index.get(difference), i]
    }
    num_index.set(nums[i], i);
  }
  return [-1, -1]
}

const one = [5,5]
const target = 10

console.log(twoSum(one, target))

File: Arrays_and_Hashing/int_sum.py
"""
AUTHOR: An Dang
DATE: 05/23/24

Solves the NeetCode Two Integer Sum problem
"""
from typing import List
def twoSum(nums: List[int], target: int) -> List[int]:
  nums_wo_num = {}
  for i in range(len(nums)):
    exclude_lst = nums.copy()
    del exclude_lst[i]
    nums_wo_num[nums[i]] = exclude_lst
    
  
  
  
def main():
  one = [3,4,0,1]
  target = 7

  twoSum(one, target)
  return 0

if __name__ == '__main__':
  main()